src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedBackingFieldSymbol.cs

```diff
@@ -122,6 +125,110 @@ public override ImmutableArray<Location> Locations
+            var binder = getAccessor.TryGetBodyBinder() ?? throw ExceptionUtilities.UnexpectedValue(getAccessor);
+            var boundGetAccessor = binder.BindMethodBody(getAccessor.SyntaxNode, BindingDiagnosticBag.Discarded);
+
+            var annotatedDiagnostics = nullableAnalyzeAndFilterDiagnostics(assumedNullableAnnotation: NullableAnnotation.Annotated);
+            if (annotatedDiagnostics.IsEmptyWithoutResolution)
+            {
+                // If the pass where the field was annotated results in no diagnostics at all, then the property is null-resilient and the not-annotated pass can be skipped.
+                annotatedDiagnostics.Free();
+                return NullableAnnotation.Annotated;
+            }
+
+            var notAnnotatedDiagnostics = nullableAnalyzeAndFilterDiagnostics(assumedNullableAnnotation: NullableAnnotation.NotAnnotated);
+            if (notAnnotatedDiagnostics.IsEmptyWithoutResolution)
+            {
+                // annotated pass had diagnostics, and not-annotated pass had no diagnostics.
+                annotatedDiagnostics.Free();
+                notAnnotatedDiagnostics.Free();
+                return NullableAnnotation.NotAnnotated;
+            }
+
+            // Both annotated and not-annotated cases had nullable warnings.
+            var notAnnotatedDiagnosticsSet = new HashSet<Diagnostic>(notAnnotatedDiagnostics.AsEnumerable(), SameDiagnosticComparer.Instance);
+            notAnnotatedDiagnostics.Free();
+
+            foreach (var diagnostic in annotatedDiagnostics.AsEnumerable())
+            {
+                if (!notAnnotatedDiagnosticsSet.Contains(diagnostic))
+                {
+                    // There is a nullable diagnostic in the pass where the field was *annotated*,
+                    // which was not present in the pass where the field is *not-annotated*. The property is not null-resilient.
+                    annotatedDiagnostics.Free();
+                    return NullableAnnotation.NotAnnotated;
+                }
+            }
+
+            annotatedDiagnostics.Free();
+            return NullableAnnotation.Annotated;
+
+            DiagnosticBag nullableAnalyzeAndFilterDiagnostics(NullableAnnotation assumedNullableAnnotation)
+            {
+                var diagnostics = DiagnosticBag.GetInstance();
+                NullableWalker.AnalyzeIfNeeded(binder, boundGetAccessor, boundGetAccessor.Syntax, diagnostics, getterNullResilienceData: (getAccessor, _property.BackingField, assumedNullableAnnotation));
+                if (diagnostics.IsEmptyWithoutResolution)
+                {
+                    return diagnostics;
+                }
+
+                var filteredDiagnostics = DiagnosticBag.GetInstance();
+                _ = DeclaringCompilation.FilterAndAppendAndFreeDiagnostics(filteredDiagnostics, ref diagnostics, cancellationToken: default);
```

Comment:
>FilterAndAppendAndFreeDiagnostics [](http://example.com/codeflow?start=41&length=33)

Does the inferred nullability of the field depend on which warnings are enabled or disabled? Shouldn't the pragmas and compiler options only control which diagnostics are reported rather than the nullability of the field?

Reply:
https://github.com/dotnet/roslyn/pull/77127#discussion_r1974484381

Reply:
Resolved in offline discussion. I think we are comfortable moving forward with the suppressions behavior currently implemented in the PR.